larpandoracontent/LArReclustering/ThreeDMultiReclusteringAlgorithm.h

```diff
@@ -0,0 +1,150 @@
+     *
+     *  @param clusters list of clusters to be deleted
+     *  @param view hit type of the clusters
+     *  @param freeCaloHits output hits of the deleted clusters
+     */
+    pandora::StatusCode FreeCaloHitsFromClusters(const pandora::ClusterList &clusters,
+                                                 const pandora::HitType &view,
+                                                 pandora::CaloHitList &freeCaloHits);
+    /**
+     *  @brief Create new pfos from the reclustered 3D clusters and original 2D hits. The original 2D hits are put into new 2D clusters
+     *         that follow the new 3D clusters,
+     *
+     *  @param clusters3D list of reclustered 3D clusters
+     *  @param viewToFreeCaloHits2D map of hit type to original 2D hits that need to be clustered
+     */
+    pandora::StatusCode BuildNewPfos(const pandora::ClusterList &clusters3D,
+                                     std::map<pandora::HitType, pandora::CaloHitList> &viewToFreeCaloHits2D);
+
+    /**
+     *  @brief Create 2D clusters following a 3D cluster
+     *
+     *  @param pCluster3D a 3D cluster
+     *  @param viewToFreeCaloHits2D map of hit type to original 2D hits that need to be clustered
+     *  @param newClusters2D output list of newly created 2D clusters
+     */
+    pandora::StatusCode Build2DClustersFrom3D(const pandora::Cluster *const pCluster3D,
+                                              std::map<pandora::HitType, pandora::CaloHitList> &viewToFreeCaloHits2D,
+                                              pandora::ClusterList &newClusters2D);
+
+    /**
+     *  @brief Add hits to their nearest cluster
+     *
+     *  @param caloHits list of 2D hits of the same hit type to be added to nearest cluster
+     *  @param clusters list of 2D clusters of the same hit type as the caloHits to have hits added to them
+     *  @param addAsIso bool for adding the hits as isolated or not
+     */
+    pandora::StatusCode MopUpCaloHits(const pandora::CaloHitList &caloHits, const pandora::ClusterList &clusters, bool addAsIso);
+
+    /**
+     *  @brief Create a new pfo from a list of clusters
+     *
+     *  @param clusters list of clusters, expect a 3D and up to 3 2D clusters.
+     */
+    pandora::StatusCode CreatePfoFromClusters(const pandora::ClusterList &clusters);
+
+    /**
+     *  @brief Get the cluster list name associated with a hit type
+     *
+     *  @param view hit type
+     *  @param listName output list name
+     */
+    pandora::StatusCode GetClusterListName(const pandora::HitType &view, std::string &listName);
+
+    pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
+
+    std::string                              m_pfoListName;       ///< Name of list of pfos to consider for reclustering
+    std::string                              m_cluster3DListName; ///< Name of list of 3D clusters that comprise the pfos
```

Comment:
Would a map of <HitType, std::string> work a bit more nicely for storing the cluster list names? I think it would at least simplify the GetClusterListName function.